src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -653,8 +651,11 @@ private static void BioWrite(SafeBioHandle bio, ReadOnlySpan<byte> buffer)
         private static Ssl.SslErrorCode GetSslError(SafeSslHandle context, int result, out Exception? innerError)
         {
             ErrorInfo lastErrno = Sys.GetLastErrorInfo(); // cache it before we make more P/Invoke calls, just in case we need it
+            return GetSslError(context, result, Ssl.SslGetError(context, result), lastErrno, out innerError);
```

Comment:
Should we go ahead and find the callers of this version (guessing SSL_read and SSL_write ops) and change them to out the error code, too?

Reply:
I would not mind, there are only few places and the pattern seems identical
Now, if we eliminate the extra call, I'm wondering if we can improve 

https://github.com/dotnet/runtime/blob/6dfbc6e2277cd8097b33e90002844364a147e74c/src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs#L652-L664

e.g. avoiding the `Crypto.ErrPeekError` and then possibly grab `ErrorInfo` only when we need to or simply capture and return errno as well.

Reply:
I also updated the read/write/renegotiate. They have same possible race condition and now we should get error back consistently with single pinvoke.

Reply:
>  I'm wondering if we can improve [the GetSslError exception-building helper]

Possibly.  The easiest way I can think of is to either use two outs, or use one long out and some local `assert`s to ensure your bit-space isn't getting trounced on.  Basically, making the helper be something like

```C#
static uint64_t GetSslError(SSL* ssl)
{
    int32_t sslErr = CryptoNative_SslGetError(ssl);

    if (sslErr == SSL_ERROR_SYSCALL)
    {
        // Something like 0x8000_0000_0000_0000
        return ERROR_SOURCE_ERRNO | errno;
    }

    return sslErr;
}
```

That's leaving the "ask Crypto.CreateOpenSslCryptographicException" part as-is, so it doesn't need to redo the work of asking if the error should be reported as OOM vs a CryptographicException.